---
title: "iSEE demo"
author: Federico Marini^[marinif@uni-mainz.de], Charlotte Soneson^[charlottesoneson@gmail.com], Kevin Rue-Albrecht^[kevinrue67@gmail.com]
output: 
    BiocStyle::html_document
vignette: >
  %\VignetteIndexEntry{iSEE demo}
  %\VignetteEngine{knitr::rmarkdown}
  %\VignetteEncoding{UTF-8}
---


```{r, include = FALSE}
knitr::opts_chunk$set(
  collapse = TRUE,
  comment = "#>"
)
suppressPackageStartupMessages({
    library(BiocStyle)
})
```

# iSEE demo

Authors:
    Federico Marini^[marinif@uni-mainz.de], Charlotte Soneson^[charlottesoneson@gmail.com], Kevin Rue-Albrecht^[kevinrue67@gmail.com]
    <br/>
Last modified: 20 July, 2023.

## Overview

### Description

Along with the topic of your workshop, include how students can expect
to spend their time. For the description may also include information
about what type of workshop it is (e.g. instructor-led live demo, lab,
lecture + lab, etc.). Instructors are strongly recommended to provide
completely worked examples for lab sessions, and a set of stand-alone
notes that can be read and understood outside of the workshop.

### Pre-requisites

List any workshop prerequisites, for example:

* Basic knowledge of R syntax
* Familiarity with the GenomicRanges class
* Familiarity with xyz vignette (provide link)

List relevant background reading for the workshop, including any
theoretical background you expect students to have.

* List any textbooks, papers, or other reading that students should be
  familiar with. Include direct links where possible.

### Participation

Describe how students will be expected to participate in the workshop.

### _R_ / _Bioconductor_ packages used

List any _R_ / _Bioconductor_ packages that will be explicitly covered.

### Time outline

An example for a 45-minute workshop:

| Activity                     | Time |
|------------------------------|------|
| Packages                     | 15m  |
| Package Development          | 15m  |
| Contributing to Bioconductor | 5m   |
| Best Practices               | 10m  |

### Workshop goals and objectives

List "big picture" student-centered workshop goals and learning
objectives. Learning goals and objectives are related, but not the
same thing. These goals and objectives will help some people to decide
whether to attend the conference for training purposes, so please make
these as precise and accurate as possible.

*Learning goals* are high-level descriptions of what
participants will learn and be able to do after the workshop is
over. *Learning objectives*, on the other hand, describe in very
specific and measurable terms specific skills or knowledge
attained. The [Bloom's Taxonomy](#bloom) may be a useful framework
for defining and describing your goals and objectives, although there
are others.

### Learning goals

Some examples:

* describe how to...
* identify methods for...
* understand the difference between...

### Learning objectives

* analyze xyz data to produce...
* create xyz plots
* evaluate xyz data for artifacts

## Loading required packages

```{r, message=FALSE}
library(macrophage)
library(DESeq2)
library(limma)
library(edgeR)
library(tximeta)
library(org.Hs.eg.db)
library(stringr)
library(SingleCellExperiment)
library(iSEE)
library(iSEEde)
```

## Preparing the macrophage data set

The data set that will be used for this demo is a bulk RNA-seq data set containing 24 samples from Alasoo, et al: "Shared genetic effects on chromatin and gene expression indicate a role for enhancer priming in immune response", published in Nature Genetics, January 2018. The 24 samples correspond to 6 different donors. For each of these we have four samples: one naive, one exposed to IFNgamma, one exposed to Salmonella, and one exposed to IFNgamma and Salmonella. The `r Biocpkg("macrophage")` Bioconductor package provides the output files resulting from running Salmon on these 24 samples. Here, we load the quantifications into R using the `r Biocpkg("tximeta")` package, and use the `r Biocpkg("DESeq2")` and `r Biocpkg("limma")` packages to perform a differential expression analysis, testing for the interaction effect between IFNgamma and Salmonella exposure, to investigate whether the effect of Salmonella exposure is different depending on whether or not the sample was also exposed to IFNgamma.

```{r, eval=FALSE}
## Set the data directory
dir <- system.file("extdata", package = "macrophage")

## Read sample annotations
coldata <- read.csv(file.path(dir, "coldata.csv"))[, c(1, 2, 3, 5)]

## Create new columns indicating, respectively whether the sample was exposed
## to IFNgamma and Salmonella
coldata$IFNg <- as.character(grepl("IFNg", coldata$condition_name))
coldata$SL1344 <- as.character(grepl("SL1344", coldata$condition_name))

## Add paths to quantification files and import data
coldata$files <- file.path(dir, "quants", coldata$names, "quant.sf.gz")
se <- tximeta(coldata = coldata, type = "salmon", dropInfReps = TRUE)

## Summarize to the gene level and add additional identifiers
seg <- summarizeToGene(se)
rownames(seg) <- str_replace(rownames(seg), "\\.\\d+$", "")
seg <- addIds(seg, "SYMBOL")
seg <- addIds(seg, "GOALL", multiVals = "list")
rownames(seg) <- scater::uniquifyFeatureNames(
    ID = rownames(seg), names = rowData(seg)$SYMBOL
)

## Create a DESeqDataSet and filter lowly expressed genes
dds <- DESeqDataSet(seg, design = ~ IFNg * SL1344)
keep <- rowSums(counts(dds)) >= 10
dds <- dds[keep,]

## Run statistical test
dds <- DESeq2::DESeq(dds)
DESeq2::resultsNames(dds)
res <- DESeq2::results(dds, name = "IFNgTRUE.SL1344TRUE", 
                       lfcThreshold = 0)

## Apply a variance stabilizing transform followed by PCA
vst <- DESeq2::varianceStabilizingTransformation(dds, blind = TRUE)
pca <- DESeq2::plotPCA(vst, intgroup = "condition_name", returnData = TRUE)

## Create a SingleCellExperiment object and include the PCA representation
sce <- as(dds, "SingleCellExperiment")
assay(sce, "vst") <- assay(vst)
SingleCellExperiment::reducedDim(sce, "PCA") <- pca[, c("PC1", "PC2")]

## Embed the contrast results obtained from DESeq2
sce <- embedContrastResults(res, sce, name = "IFNgTRUE.SL1344TRUE.DESeq2")

## Run limma-trend
dge <- tximeta::makeDGEList(seg)
dge <- dge[rownames(dds), ]
logCPM <- edgeR::cpm(dge, log = TRUE, prior.count = 3)
design <- model.matrix(~ IFNg * SL1344, data = dge$samples)
fit <- limma::lmFit(logCPM, design = design)
fit <- eBayes(fit, trend = TRUE)
tt <- topTable(fit, coef = ncol(design), number = Inf, sort.by = "none")
sce <- embedContrastResults(tt, sce, name = "IFNgTRUE.SL1344TRUE.limma", class = "limma")
```


