<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>iSEEing is believing: the iSEE package for efficient, interactive, and reproducible exploration of data • iSEEDemoEuroBioC2023</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link href="https://cdnjs.cloudflare.com/ajax/libs/bootswatch/3.4.0/flatly/bootstrap.min.css" rel="stylesheet" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="iSEEing is believing: the iSEE package for efficient, interactive, and reproducible exploration of data">
<meta property="og:description" content="iSEEDemoEuroBioC2023">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    

    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">iSEEDemoEuroBioC2023</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">2.0.0</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/HOWTO_BUILD_WORKSHOP.html">How To Build A Workshop Package</a>
    </li>
    <li>
      <a href="../articles/iSEEdemo.html">iSEEing is believing: the iSEE package for efficient, interactive, and reproducible exploration of data</a>
    </li>
  </ul>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/iSEE/iSEEDemoEuroBioC2023" class="external-link">
    <span class="fa fa-github"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>iSEEing is believing: the iSEE package for
efficient, interactive, and reproducible exploration of data</h1>
                        <h4 data-toc-skip class="author">Federico
Marini<a href="#fn1" class="footnote-ref" id="fnref1"><sup>1</sup></a>,
Charlotte Soneson<a href="#fn2" class="footnote-ref" id="fnref2"><sup>2</sup></a>, Kevin Rue-Albrecht<a href="#fn3" class="footnote-ref" id="fnref3"><sup>3</sup></a>
</h4>
            
      
      <small class="dont-index">Source: <a href="https://github.com/iSEE/iSEEDemoEuroBioC2023/blob/HEAD/vignettes/iSEEdemo.Rmd" class="external-link"><code>vignettes/iSEEdemo.Rmd</code></a></small>
      <div class="hidden name"><code>iSEEdemo.Rmd</code></div>

    </div>

    
    
<div class="section level2">
<h2 id="overview">Overview<a class="anchor" aria-label="anchor" href="#overview"></a>
</h2>
<div class="section level3">
<h3 id="description">Description<a class="anchor" aria-label="anchor" href="#description"></a>
</h3>
<p>This package demo provides an overview of recent features added to
the “iSEE universe”, the set of packages revolving around and extending
the <em><a href="https://bioconductor.org/packages/3.18/iSEE" class="external-link">iSEE</a></em>
Bioconductor package.</p>
</div>
<div class="section level3">
<h3 id="pre-requisites">Pre-requisites<a class="anchor" aria-label="anchor" href="#pre-requisites"></a>
</h3>
<ul>
<li>Familiarity with the SummarizedExperiment class</li>
<li>(Optional, but helpful) Basic familiarity with the iSEE package</li>
</ul>
</div>
<div class="section level3">
<h3 id="participation">Participation<a class="anchor" aria-label="anchor" href="#participation"></a>
</h3>
<p>Participants will be able to follow along in the live coding parts of
the demo.</p>
</div>
<div class="section level3">
<h3 id="r-bioconductor-packages-used">
<em>R</em> / <em>Bioconductor</em> packages used<a class="anchor" aria-label="anchor" href="#r-bioconductor-packages-used"></a>
</h3>
<ul>
<li>iSEE</li>
<li>iSEEde</li>
<li>iSEEindex</li>
<li>iSEEpathways</li>
<li>iSEEhub</li>
</ul>
</div>
<div class="section level3">
<h3 id="time-outline">Time outline<a class="anchor" aria-label="anchor" href="#time-outline"></a>
</h3>
<table class="table">
<thead><tr class="header">
<th>Activity</th>
<th>Time</th>
</tr></thead>
<tbody>
<tr class="odd">
<td>Overview of data set</td>
<td>10m</td>
</tr>
<tr class="even">
<td>Overview of iSEE</td>
<td>10m</td>
</tr>
<tr class="odd">
<td>iSEEde + iSEEpathways</td>
<td>10m</td>
</tr>
<tr class="even">
<td>iSEEhub + iSEEindex</td>
<td>15m</td>
</tr>
</tbody>
</table>
</div>
<div class="section level3">
<h3 id="workshop-goals-and-objectives">Workshop goals and objectives<a class="anchor" aria-label="anchor" href="#workshop-goals-and-objectives"></a>
</h3>
<p>List “big picture” student-centered workshop goals and learning
objectives. Learning goals and objectives are related, but not the same
thing. These goals and objectives will help some people to decide
whether to attend the conference for training purposes, so please make
these as precise and accurate as possible.</p>
<p><em>Learning goals</em> are high-level descriptions of what
participants will learn and be able to do after the workshop is over.
<em>Learning objectives</em>, on the other hand, describe in very
specific and measurable terms specific skills or knowledge attained. The
<a href="#bloom">Bloom’s Taxonomy</a> may be a useful framework for
defining and describing your goals and objectives, although there are
others.</p>
</div>
<div class="section level3">
<h3 id="learning-goals">Learning goals<a class="anchor" aria-label="anchor" href="#learning-goals"></a>
</h3>
<p>Some examples:</p>
<ul>
<li>describe how to…</li>
<li>identify methods for…</li>
<li>understand the difference between…</li>
</ul>
</div>
<div class="section level3">
<h3 id="learning-objectives">Learning objectives<a class="anchor" aria-label="anchor" href="#learning-objectives"></a>
</h3>
<ul>
<li>analyze xyz data to produce…</li>
<li>create xyz plots</li>
<li>evaluate xyz data for artifacts</li>
</ul>
</div>
</div>
<div class="section level2">
<h2 id="loading-required-packages">Loading required packages<a class="anchor" aria-label="anchor" href="#loading-required-packages"></a>
</h2>
<p>We start by loading the packages that will be used during the
demo.</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va">macrophage</span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/thelovelab/DESeq2" class="external-link">DESeq2</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://bioinf.wehi.edu.au/limma/" class="external-link">limma</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://bioinf.wehi.edu.au/edgeR/" class="external-link">edgeR</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/thelovelab/tximeta" class="external-link">tximeta</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va">org.Hs.eg.db</span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://stringr.tidyverse.org" class="external-link">stringr</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va">SingleCellExperiment</span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/iSEE/iSEE" class="external-link">iSEE</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/iSEE/iSEEde" class="external-link">iSEEde</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/iSEE/iSEEpathways" class="external-link">iSEEpathways</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/iSEE/iSEEhub" class="external-link">iSEEhub</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/iSEE/iSEEindex" class="external-link">iSEEindex</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va">org.Hs.eg.db</span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/ctlab/fgsea/" class="external-link">fgsea</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va">GO.db</span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va">ExperimentHub</span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va">DuoClustering2018</span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va">BiocFileCache</span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="http://bioconductor.org/packages/scater/" class="external-link">scater</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/vubiostat/r-yaml/" class="external-link">yaml</a></span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="loading-the-macrophage-data-set">Loading the macrophage data set<a class="anchor" aria-label="anchor" href="#loading-the-macrophage-data-set"></a>
</h2>
<p>The data set that will be used for this demo is a bulk RNA-seq data
set containing 24 samples from Alasoo, et al: “Shared genetic effects on
chromatin and gene expression indicate a role for enhancer priming in
immune response”, published in Nature Genetics, January 2018. The 24
samples correspond to 6 different donors. For each of these we have four
samples: one naive, one exposed to IFNgamma, one exposed to Salmonella,
and one exposed to IFNgamma and Salmonella. The <em><a href="https://bioconductor.org/packages/3.18/macrophage" class="external-link">macrophage</a></em>
Bioconductor package provides the output files resulting from running
Salmon on these 24 samples. Here, we load the quantifications into R
using the <em><a href="https://bioconductor.org/packages/3.18/tximeta" class="external-link">tximeta</a></em>
package.</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## Set the data directory</span></span>
<span><span class="va">dir</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/system.file.html" class="external-link">system.file</a></span><span class="op">(</span><span class="st">"extdata"</span>, package <span class="op">=</span> <span class="st">"macrophage"</span><span class="op">)</span></span>
<span></span>
<span><span class="co">## Read sample annotations</span></span>
<span><span class="va">coldata</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/utils/read.table.html" class="external-link">read.csv</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/file.path.html" class="external-link">file.path</a></span><span class="op">(</span><span class="va">dir</span>, <span class="st">"coldata.csv"</span><span class="op">)</span><span class="op">)</span><span class="op">[</span>, <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">1</span>, <span class="fl">2</span>, <span class="fl">3</span>, <span class="fl">5</span><span class="op">)</span><span class="op">]</span></span>
<span></span>
<span><span class="co">## Create new columns indicating, respectively whether the sample was exposed</span></span>
<span><span class="co">## to IFNgamma and Salmonella</span></span>
<span><span class="va">coldata</span><span class="op">$</span><span class="va">IFNg</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/character.html" class="external-link">as.character</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/grep.html" class="external-link">grepl</a></span><span class="op">(</span><span class="st">"IFNg"</span>, <span class="va">coldata</span><span class="op">$</span><span class="va">condition_name</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">coldata</span><span class="op">$</span><span class="va">SL1344</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/character.html" class="external-link">as.character</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/grep.html" class="external-link">grepl</a></span><span class="op">(</span><span class="st">"SL1344"</span>, <span class="va">coldata</span><span class="op">$</span><span class="va">condition_name</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="co">## Add paths to quantification files and import data</span></span>
<span><span class="va">coldata</span><span class="op">$</span><span class="va">files</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/file.path.html" class="external-link">file.path</a></span><span class="op">(</span><span class="va">dir</span>, <span class="st">"quants"</span>, <span class="va">coldata</span><span class="op">$</span><span class="va">names</span>, <span class="st">"quant.sf.gz"</span><span class="op">)</span></span>
<span><span class="va">se</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/tximeta/man/tximeta.html" class="external-link">tximeta</a></span><span class="op">(</span>coldata <span class="op">=</span> <span class="va">coldata</span>, type <span class="op">=</span> <span class="st">"salmon"</span>, dropInfReps <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span>
<span><span class="co">#&gt; importing quantifications</span></span>
<span><span class="co">#&gt; reading in files with read.delim (install 'readr' package for speed up)</span></span>
<span><span class="co">#&gt; 1 2 3 4 5 6 7 8 9 10 11 12 13 14 15 16 17 18 19 20 21 22 23 24 </span></span>
<span><span class="co">#&gt; found matching transcriptome:</span></span>
<span><span class="co">#&gt; [ GENCODE - Homo sapiens - release 29 ]</span></span>
<span><span class="co">#&gt; useHub=TRUE: checking for TxDb via 'AnnotationHub'</span></span>
<span><span class="co">#&gt; found matching TxDb via 'AnnotationHub'</span></span>
<span><span class="co">#&gt; loading from cache</span></span>
<span><span class="co">#&gt; Loading required package: GenomicFeatures</span></span>
<span><span class="co">#&gt; generating transcript ranges</span></span>
<span></span>
<span><span class="co">## Summarize to the gene level and add additional identifiers</span></span>
<span><span class="va">seg</span> <span class="op">&lt;-</span> <span class="fu">summarizeToGene</span><span class="op">(</span><span class="va">se</span><span class="op">)</span></span>
<span><span class="co">#&gt; loading existing TxDb created: 2023-09-05 20:21:15</span></span>
<span><span class="co">#&gt; obtaining transcript-to-gene mapping from database</span></span>
<span><span class="co">#&gt; generating gene ranges</span></span>
<span><span class="co">#&gt; gene ranges assigned by total range of isoforms, see `assignRanges`</span></span>
<span><span class="co">#&gt; summarizing abundance</span></span>
<span><span class="co">#&gt; summarizing counts</span></span>
<span><span class="co">#&gt; summarizing length</span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/row_colnames.html" class="external-link">rownames</a></span><span class="op">(</span><span class="va">seg</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://stringr.tidyverse.org/reference/str_replace.html" class="external-link">str_replace</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/row_colnames.html" class="external-link">rownames</a></span><span class="op">(</span><span class="va">seg</span><span class="op">)</span>, <span class="st">"\\.\\d+$"</span>, <span class="st">""</span><span class="op">)</span></span>
<span><span class="va">seg</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/tximeta/man/addIds.html" class="external-link">addIds</a></span><span class="op">(</span><span class="va">seg</span>, <span class="st">"SYMBOL"</span><span class="op">)</span></span>
<span><span class="co">#&gt; it appears the rows are gene IDs, setting 'gene' to TRUE</span></span>
<span><span class="co">#&gt; mapping to new IDs using org.Hs.eg.db</span></span>
<span><span class="co">#&gt; if all matching IDs are desired, and '1:many mappings' are reported,</span></span>
<span><span class="co">#&gt; set multiVals='list' to obtain all the matching IDs</span></span>
<span><span class="co">#&gt; 'select()' returned 1:many mapping between keys and columns</span></span>
<span><span class="va">seg</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/tximeta/man/addIds.html" class="external-link">addIds</a></span><span class="op">(</span><span class="va">seg</span>, <span class="st">"GOALL"</span>, multiVals <span class="op">=</span> <span class="st">"list"</span><span class="op">)</span></span>
<span><span class="co">#&gt; it appears the rows are gene IDs, setting 'gene' to TRUE</span></span>
<span><span class="co">#&gt; mapping to new IDs using org.Hs.eg.db</span></span>
<span><span class="co">#&gt; if all matching IDs are desired, and '1:many mappings' are reported,</span></span>
<span><span class="co">#&gt; set multiVals='list' to obtain all the matching IDs</span></span>
<span><span class="co">#&gt; 'select()' returned 1:many mapping between keys and columns</span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/row_colnames.html" class="external-link">rownames</a></span><span class="op">(</span><span class="va">seg</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu">scater</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/scuttle/man/uniquifyFeatureNames.html" class="external-link">uniquifyFeatureNames</a></span><span class="op">(</span></span>
<span>    ID <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/row_colnames.html" class="external-link">rownames</a></span><span class="op">(</span><span class="va">seg</span><span class="op">)</span>, names <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html" class="external-link">rowData</a></span><span class="op">(</span><span class="va">seg</span><span class="op">)</span><span class="op">$</span><span class="va">SYMBOL</span></span>
<span><span class="op">)</span></span>
<span></span>
<span><span class="co">## Create a DESeqDataSet and filter lowly expressed genes</span></span>
<span><span class="va">dds</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/DESeq2/man/DESeqDataSet.html" class="external-link">DESeqDataSet</a></span><span class="op">(</span><span class="va">seg</span>, design <span class="op">=</span> <span class="op">~</span> <span class="va">IFNg</span> <span class="op">*</span> <span class="va">SL1344</span><span class="op">)</span></span>
<span><span class="co">#&gt; using counts and average transcript lengths from tximeta</span></span>
<span><span class="co">#&gt; Warning in DESeqDataSet(seg, design = ~IFNg * SL1344): some variables in design</span></span>
<span><span class="co">#&gt; formula are characters, converting to factors</span></span>
<span><span class="va">keep</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/colSums.html" class="external-link">rowSums</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/dge.html" class="external-link">counts</a></span><span class="op">(</span><span class="va">dds</span><span class="op">)</span><span class="op">)</span> <span class="op">&gt;=</span> <span class="fl">10</span></span>
<span><span class="va">dds</span> <span class="op">&lt;-</span> <span class="va">dds</span><span class="op">[</span><span class="va">keep</span>,<span class="op">]</span></span>
<span></span>
<span><span class="co">## Apply a variance stabilizing transform followed by PCA</span></span>
<span><span class="va">vst</span> <span class="op">&lt;-</span> <span class="fu">DESeq2</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/DESeq2/man/varianceStabilizingTransformation.html" class="external-link">varianceStabilizingTransformation</a></span><span class="op">(</span><span class="va">dds</span>, blind <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span>
<span><span class="co">#&gt; using 'avgTxLength' from assays(dds), correcting for library size</span></span>
<span><span class="va">pca</span> <span class="op">&lt;-</span> <span class="fu">DESeq2</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/plotPCA.html" class="external-link">plotPCA</a></span><span class="op">(</span><span class="va">vst</span>, intgroup <span class="op">=</span> <span class="st">"condition_name"</span>, returnData <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span>
<span></span>
<span><span class="co">## Create a SingleCellExperiment object and include the PCA representation</span></span>
<span><span class="va">sce</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/methods/as.html" class="external-link">as</a></span><span class="op">(</span><span class="va">dds</span>, <span class="st">"SingleCellExperiment"</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html" class="external-link">assay</a></span><span class="op">(</span><span class="va">sce</span>, <span class="st">"vst"</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html" class="external-link">assay</a></span><span class="op">(</span><span class="va">vst</span><span class="op">)</span></span>
<span><span class="fu">SingleCellExperiment</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/SingleCellExperiment/man/reducedDims.html" class="external-link">reducedDim</a></span><span class="op">(</span><span class="va">sce</span>, <span class="st">"PCA"</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="va">pca</span><span class="op">[</span>, <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"PC1"</span>, <span class="st">"PC2"</span><span class="op">)</span><span class="op">]</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="performing-differential-expression-analysis">Performing differential expression analysis<a class="anchor" aria-label="anchor" href="#performing-differential-expression-analysis"></a>
</h2>
<p>Below, we will illustrate how to use the <em><a href="https://bioconductor.org/packages/3.18/iSEEde" class="external-link">iSEEde</a></em> and
<em><a href="https://bioconductor.org/packages/3.18/iSEEpathways" class="external-link">iSEEpathways</a></em>
packages to visualize results from differential expression analyses.
Thus, we first apply the <em><a href="https://bioconductor.org/packages/3.18/DESeq2" class="external-link">DESeq2</a></em> and
<em><a href="https://bioconductor.org/packages/3.18/limma" class="external-link">limma</a></em>
packages to perform differential expression analysis. We test the
interaction effect between IFNgamma and Salmonella exposure, to
investigate whether the effect of Salmonella exposure is different
depending on whether or not the sample was also exposed to IFNgamma.</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## Run DESeq2</span></span>
<span><span class="va">dds</span> <span class="op">&lt;-</span> <span class="fu">DESeq2</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/DESeq2/man/DESeq.html" class="external-link">DESeq</a></span><span class="op">(</span><span class="va">dds</span><span class="op">)</span></span>
<span><span class="co">#&gt; estimating size factors</span></span>
<span><span class="co">#&gt; using 'avgTxLength' from assays(dds), correcting for library size</span></span>
<span><span class="co">#&gt; estimating dispersions</span></span>
<span><span class="co">#&gt; gene-wise dispersion estimates</span></span>
<span><span class="co">#&gt; mean-dispersion relationship</span></span>
<span><span class="co">#&gt; final dispersion estimates</span></span>
<span><span class="co">#&gt; fitting model and testing</span></span>
<span><span class="fu">DESeq2</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/DESeq2/man/results.html" class="external-link">resultsNames</a></span><span class="op">(</span><span class="va">dds</span><span class="op">)</span></span>
<span><span class="co">#&gt; [1] "Intercept"            "IFNg_TRUE_vs_FALSE"   "SL1344_TRUE_vs_FALSE"</span></span>
<span><span class="co">#&gt; [4] "IFNgTRUE.SL1344TRUE"</span></span>
<span><span class="va">res</span> <span class="op">&lt;-</span> <span class="fu">DESeq2</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/DESeq2/man/results.html" class="external-link">results</a></span><span class="op">(</span><span class="va">dds</span>, name <span class="op">=</span> <span class="st">"IFNgTRUE.SL1344TRUE"</span>, </span>
<span>                       lfcThreshold <span class="op">=</span> <span class="fl">0</span><span class="op">)</span></span>
<span></span>
<span><span class="co">## Embed the contrast results obtained from DESeq2 into the SCE</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/stopifnot.html" class="external-link">stopifnot</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/row_colnames.html" class="external-link">rownames</a></span><span class="op">(</span><span class="va">res</span><span class="op">)</span> <span class="op">==</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/row_colnames.html" class="external-link">rownames</a></span><span class="op">(</span><span class="va">sce</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">sce</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/iSEEde/man/utils-SummarizedExperiment.html" class="external-link">embedContrastResults</a></span><span class="op">(</span><span class="va">res</span>, <span class="va">sce</span>, name <span class="op">=</span> <span class="st">"IFNgTRUE.SL1344TRUE.DESeq2"</span><span class="op">)</span></span>
<span></span>
<span><span class="co">## Run limma-trend</span></span>
<span><span class="va">dge</span> <span class="op">&lt;-</span> <span class="fu">tximeta</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/tximeta/man/makeDGEList.html" class="external-link">makeDGEList</a></span><span class="op">(</span><span class="va">seg</span><span class="op">)</span></span>
<span><span class="va">dge</span> <span class="op">&lt;-</span> <span class="va">dge</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/row_colnames.html" class="external-link">rownames</a></span><span class="op">(</span><span class="va">dds</span><span class="op">)</span>, <span class="op">]</span></span>
<span><span class="va">logCPM</span> <span class="op">&lt;-</span> <span class="fu">edgeR</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/edgeR/man/cpm.html" class="external-link">cpm</a></span><span class="op">(</span><span class="va">dge</span>, log <span class="op">=</span> <span class="cn">TRUE</span>, prior.count <span class="op">=</span> <span class="fl">3</span><span class="op">)</span></span>
<span><span class="va">design</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/stats/model.matrix.html" class="external-link">model.matrix</a></span><span class="op">(</span><span class="op">~</span> <span class="va">IFNg</span> <span class="op">*</span> <span class="va">SL1344</span>, data <span class="op">=</span> <span class="va">dge</span><span class="op">$</span><span class="va">samples</span><span class="op">)</span></span>
<span><span class="va">fit</span> <span class="op">&lt;-</span> <span class="fu">limma</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/limma/man/lmFit.html" class="external-link">lmFit</a></span><span class="op">(</span><span class="va">logCPM</span>, design <span class="op">=</span> <span class="va">design</span><span class="op">)</span></span>
<span><span class="va">fit</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/limma/man/ebayes.html" class="external-link">eBayes</a></span><span class="op">(</span><span class="va">fit</span>, trend <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span>
<span><span class="va">tt</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/limma/man/toptable.html" class="external-link">topTable</a></span><span class="op">(</span><span class="va">fit</span>, coef <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/nrow.html" class="external-link">ncol</a></span><span class="op">(</span><span class="va">design</span><span class="op">)</span>, number <span class="op">=</span> <span class="cn">Inf</span>, sort.by <span class="op">=</span> <span class="st">"none"</span><span class="op">)</span></span>
<span></span>
<span><span class="co">## Embed the results obtained from limma into the SCE</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/stopifnot.html" class="external-link">stopifnot</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/row_colnames.html" class="external-link">rownames</a></span><span class="op">(</span><span class="va">tt</span><span class="op">)</span> <span class="op">==</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/row_colnames.html" class="external-link">rownames</a></span><span class="op">(</span><span class="va">sce</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">sce</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/iSEEde/man/utils-SummarizedExperiment.html" class="external-link">embedContrastResults</a></span><span class="op">(</span><span class="va">tt</span>, <span class="va">sce</span>, name <span class="op">=</span> <span class="st">"IFNgTRUE.SL1344TRUE.limma"</span>, class <span class="op">=</span> <span class="st">"limma"</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="perform-pathway-analysis">Perform pathway analysis<a class="anchor" aria-label="anchor" href="#perform-pathway-analysis"></a>
</h2>
<p>We also run <em><a href="https://bioconductor.org/packages/3.18/fgsea" class="external-link">fgsea</a></em> to
perform a gene set enrichment analysis of GO terms, based on the
<code>DESeq2</code> results obtained above.</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## Extract GO term composition </span></span>
<span><span class="va">pathways</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/AnnotationDbi/man/AnnotationDb-class.html" class="external-link">select</a></span><span class="op">(</span><span class="va">org.Hs.eg.db</span>, <span class="fu"><a href="https://rdrr.io/pkg/AnnotationDbi/man/AnnotationDb-class.html" class="external-link">keys</a></span><span class="op">(</span><span class="va">org.Hs.eg.db</span>, <span class="st">"SYMBOL"</span><span class="op">)</span>, <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"GOALL"</span><span class="op">)</span>, </span>
<span>                   keytype <span class="op">=</span> <span class="st">"SYMBOL"</span><span class="op">)</span></span>
<span><span class="co">#&gt; 'select()' returned 1:many mapping between keys and columns</span></span>
<span><span class="va">pathways</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/subset.html" class="external-link">subset</a></span><span class="op">(</span><span class="va">pathways</span>, <span class="va">ONTOLOGYALL</span> <span class="op">==</span> <span class="st">"BP"</span><span class="op">)</span></span>
<span><span class="va">pathways</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/unique.html" class="external-link">unique</a></span><span class="op">(</span><span class="va">pathways</span><span class="op">[</span>, <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"SYMBOL"</span>, <span class="st">"GOALL"</span><span class="op">)</span><span class="op">]</span><span class="op">)</span></span>
<span><span class="va">pathways</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/S4Vectors/man/splitAsList.html" class="external-link">split</a></span><span class="op">(</span><span class="va">pathways</span><span class="op">$</span><span class="va">SYMBOL</span>, <span class="va">pathways</span><span class="op">$</span><span class="va">GOALL</span><span class="op">)</span></span>
<span><span class="va">len_pathways</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/lengths.html" class="external-link">lengths</a></span><span class="op">(</span><span class="va">pathways</span><span class="op">)</span></span>
<span><span class="va">pathways</span> <span class="op">&lt;-</span> <span class="va">pathways</span><span class="op">[</span><span class="va">len_pathways</span> <span class="op">&gt;</span> <span class="fl">15</span> <span class="op">&amp;</span> <span class="va">len_pathways</span> <span class="op">&lt;</span> <span class="fl">200</span><span class="op">]</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/length.html" class="external-link">length</a></span><span class="op">(</span><span class="va">pathways</span><span class="op">)</span></span>
<span><span class="co">#&gt; [1] 4454</span></span>
<span></span>
<span><span class="co">## Get test statistics and apply fgsea</span></span>
<span><span class="va">feature_stats</span> <span class="op">&lt;-</span> <span class="va">res</span><span class="op">$</span><span class="va">stat</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">feature_stats</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/row_colnames.html" class="external-link">rownames</a></span><span class="op">(</span><span class="va">res</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/Random.html" class="external-link">set.seed</a></span><span class="op">(</span><span class="fl">42</span><span class="op">)</span></span>
<span><span class="va">fgseaRes</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/fgsea/man/fgsea.html" class="external-link">fgsea</a></span><span class="op">(</span>pathways <span class="op">=</span> <span class="va">pathways</span>, </span>
<span>                  stats <span class="op">=</span> <span class="va">feature_stats</span>,</span>
<span>                  minSize <span class="op">=</span> <span class="fl">15</span>,</span>
<span>                  maxSize <span class="op">=</span> <span class="fl">200</span><span class="op">)</span></span>
<span><span class="va">fgseaRes</span> <span class="op">&lt;-</span> <span class="va">fgseaRes</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/order.html" class="external-link">order</a></span><span class="op">(</span><span class="va">pval</span><span class="op">)</span>, <span class="op">]</span></span>
<span></span>
<span><span class="co">## Embed pathway analysis results in the SCE object</span></span>
<span><span class="va">sce</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/iSEEpathways/man/utils-SummarizedExperiment.html" class="external-link">embedPathwaysResults</a></span><span class="op">(</span><span class="va">fgseaRes</span>, <span class="va">sce</span>, name <span class="op">=</span> <span class="st">"IFNgTRUE.SL1344TRUE.limma.fgsea"</span>, </span>
<span>                            class <span class="op">=</span> <span class="st">"fgsea"</span>, pathwayType <span class="op">=</span> <span class="st">"GO"</span>, pathwaysList <span class="op">=</span> <span class="va">pathways</span>, </span>
<span>                            featuresStats <span class="op">=</span> <span class="va">feature_stats</span><span class="op">)</span></span>
<span></span>
<span><span class="co">## Add details about the GO terms</span></span>
<span><span class="co">## These will be displayed in the application</span></span>
<span><span class="va">go_details</span> <span class="op">&lt;-</span> <span class="kw">function</span><span class="op">(</span><span class="va">x</span><span class="op">)</span> <span class="op">{</span></span>
<span>    <span class="va">info</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/AnnotationDbi/man/AnnotationDb-class.html" class="external-link">select</a></span><span class="op">(</span><span class="va">GO.db</span>, <span class="va">x</span>, <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"TERM"</span>, <span class="st">"ONTOLOGY"</span>, <span class="st">"DEFINITION"</span><span class="op">)</span>, <span class="st">"GOID"</span><span class="op">)</span></span>
<span>    <span class="va">html</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span><span class="fu">p</span><span class="op">(</span><span class="fu">strong</span><span class="op">(</span><span class="va">info</span><span class="op">$</span><span class="va">GOID</span><span class="op">)</span>, <span class="st">":"</span>, <span class="va">info</span><span class="op">$</span><span class="va">TERM</span>, <span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste0</a></span><span class="op">(</span><span class="st">"("</span>, <span class="va">info</span><span class="op">$</span><span class="va">ONTOLOGY</span>, <span class="st">")"</span><span class="op">)</span><span class="op">)</span><span class="op">)</span></span>
<span>    <span class="kw">if</span> <span class="op">(</span><span class="op">!</span><span class="fu"><a href="https://rdrr.io/r/base/NA.html" class="external-link">is.na</a></span><span class="op">(</span><span class="va">info</span><span class="op">$</span><span class="va">DEFINITION</span><span class="op">)</span><span class="op">)</span> <span class="op">{</span></span>
<span>        <span class="va">html</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/append.html" class="external-link">append</a></span><span class="op">(</span><span class="va">html</span>, <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span><span class="fu">p</span><span class="op">(</span><span class="va">info</span><span class="op">$</span><span class="va">DEFINITION</span><span class="op">)</span><span class="op">)</span><span class="op">)</span></span>
<span>    <span class="op">}</span></span>
<span>    <span class="fu">tagList</span><span class="op">(</span><span class="va">html</span><span class="op">)</span></span>
<span><span class="op">}</span></span>
<span></span>
<span><span class="co">## Define the mapping from GO terms to gene IDs</span></span>
<span><span class="va">map_GO</span> <span class="op">&lt;-</span> <span class="kw">function</span><span class="op">(</span><span class="va">pathway_id</span>, <span class="va">se</span><span class="op">)</span> <span class="op">{</span></span>
<span>    <span class="va">pathway_symbol</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/AnnotationDbi/man/AnnotationDb-class.html" class="external-link">mapIds</a></span><span class="op">(</span><span class="va">org.Hs.eg.db</span>, <span class="va">pathway_id</span>, <span class="st">"SYMBOL"</span>, </span>
<span>                             keytype <span class="op">=</span> <span class="st">"GOALL"</span>, multiVals <span class="op">=</span> <span class="st">"CharacterList"</span><span class="op">)</span><span class="op">[[</span><span class="va">pathway_id</span><span class="op">]</span><span class="op">]</span></span>
<span>    <span class="va">pathway_rownames</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/row_colnames.html" class="external-link">rownames</a></span><span class="op">(</span><span class="va">se</span><span class="op">)</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html" class="external-link">rowData</a></span><span class="op">(</span><span class="va">se</span><span class="op">)</span><span class="op">$</span><span class="va">SYMBOL</span> <span class="op"><a href="https://rdrr.io/pkg/BiocGenerics/man/match.html" class="external-link">%in%</a></span> <span class="va">pathway_symbol</span><span class="op">]</span></span>
<span>    <span class="va">pathway_rownames</span></span>
<span><span class="op">}</span></span>
<span></span>
<span><span class="co">## Register the pathway mapping information into the SCE object</span></span>
<span><span class="va">sce</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/iSEE/man/registerAppOptions.html" class="external-link">registerAppOptions</a></span><span class="op">(</span><span class="va">sce</span>, Pathways.map.functions <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span>GO <span class="op">=</span> <span class="va">map_GO</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">sce</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/iSEE/man/registerAppOptions.html" class="external-link">registerAppOptions</a></span><span class="op">(</span><span class="va">sce</span>, PathwaysTable.select.details <span class="op">=</span> <span class="va">go_details</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="a-quick-overview-of-isee">A quick overview of <code>iSEE</code><a class="anchor" aria-label="anchor" href="#a-quick-overview-of-isee"></a>
</h2>
<p><em><a href="https://bioconductor.org/packages/3.18/iSEE" class="external-link">iSEE</a></em> is a
Bioconductor package, based on <em><a href="https://CRAN.R-project.org/package=shiny" class="external-link">shiny</a></em>, that
allows the user to create an interactive interface for exploring their
data. The input should be an object of the class
<code>SummarizedExperiment</code>, or any class extending
<code>SummarizedExperiment</code> (e.g.,
<code>SingleCellExperiment</code>, <code>DESeqDataSet</code>). Launching
an application is as simple as:</p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/pkg/iSEE/man/iSEE.html" class="external-link">iSEE</a></span><span class="op">(</span><span class="va">sce</span><span class="op">)</span></span></code></pre></div>
<p>For a more extensive description of the different parts of the
interface, we refer to the <a href="overview.html">overview</a>
vignette.</p>
</div>
<div class="section level2">
<h2 id="iseede-and-iseepathways">iSEEde and iSEEpathways<a class="anchor" aria-label="anchor" href="#iseede-and-iseepathways"></a>
</h2>
<p><em><a href="https://bioconductor.org/packages/3.18/iSEEde" class="external-link">iSEEde</a></em> and
<em><a href="https://bioconductor.org/packages/3.18/iSEEpathways" class="external-link">iSEEpathways</a></em>
are two new Bioconductor packages that provide <em><a href="https://bioconductor.org/packages/3.18/iSEE" class="external-link">iSEE</a></em> panels
specifically aimed towards exploration of differential expression and
pathway analysis results. More precisely, <em><a href="https://bioconductor.org/packages/3.18/iSEEde" class="external-link">iSEEde</a></em>
provides the <code>VolcanoPlot</code>, <code>MAPlot</code>,
<code>LogFCLogFCPlot</code> and <code>DETable</code> panels. These
panels can be configured to extract data that was added via the
<code><a href="https://rdrr.io/pkg/iSEEde/man/utils-SummarizedExperiment.html" class="external-link">embedContrastResults()</a></code> function above. Let’s look at an
example:</p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">app</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/iSEE/man/iSEE.html" class="external-link">iSEE</a></span><span class="op">(</span><span class="va">sce</span>, initial <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span></span>
<span>    <span class="fu"><a href="https://rdrr.io/pkg/iSEEde/man/DETable-class.html" class="external-link">DETable</a></span><span class="op">(</span>ContrastName <span class="op">=</span> <span class="st">"IFNgTRUE.SL1344TRUE.DESeq2"</span>, </span>
<span>            HiddenColumns <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"baseMean"</span>, <span class="st">"lfcSE"</span>, <span class="st">"stat"</span><span class="op">)</span><span class="op">)</span>,</span>
<span>    <span class="fu"><a href="https://rdrr.io/pkg/iSEEde/man/VolcanoPlot-class.html" class="external-link">VolcanoPlot</a></span><span class="op">(</span>ContrastName <span class="op">=</span> <span class="st">"IFNgTRUE.SL1344TRUE.DESeq2"</span><span class="op">)</span>,</span>
<span>    <span class="fu"><a href="https://rdrr.io/pkg/iSEEde/man/MAPlot-class.html" class="external-link">MAPlot</a></span><span class="op">(</span>ContrastName <span class="op">=</span> <span class="st">"IFNgTRUE.SL1344TRUE.DESeq2"</span><span class="op">)</span></span>
<span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">app</span></span></code></pre></div>
<p>Note how it is easy to switch to a different contrast in any of the
panels.</p>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">app</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/iSEE/man/iSEE.html" class="external-link">iSEE</a></span><span class="op">(</span><span class="va">sce</span>, initial <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span></span>
<span>    <span class="fu"><a href="https://rdrr.io/pkg/iSEEde/man/DETable-class.html" class="external-link">DETable</a></span><span class="op">(</span>ContrastName <span class="op">=</span> <span class="st">"IFNgTRUE.SL1344TRUE.DESeq2"</span>, </span>
<span>            HiddenColumns <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"baseMean"</span>, <span class="st">"lfcSE"</span>, <span class="st">"stat"</span><span class="op">)</span><span class="op">)</span>,</span>
<span>    <span class="fu"><a href="https://rdrr.io/pkg/iSEEde/man/VolcanoPlot-class.html" class="external-link">VolcanoPlot</a></span><span class="op">(</span>ContrastName <span class="op">=</span> <span class="st">"IFNgTRUE.SL1344TRUE.DESeq2"</span><span class="op">)</span>,</span>
<span>    <span class="fu"><a href="https://rdrr.io/pkg/iSEEde/man/MAPlot-class.html" class="external-link">MAPlot</a></span><span class="op">(</span>ContrastName <span class="op">=</span> <span class="st">"IFNgTRUE.SL1344TRUE.DESeq2"</span><span class="op">)</span>,</span>
<span>    <span class="fu"><a href="https://rdrr.io/pkg/iSEEpathways/man/PathwaysTable-class.html" class="external-link">PathwaysTable</a></span><span class="op">(</span>ResultName <span class="op">=</span> <span class="st">"IFNgTRUE.SL1344TRUE.limma.fgsea"</span>, </span>
<span>                  Selected <span class="op">=</span> <span class="st">"GO:0046324"</span><span class="op">)</span>,</span>
<span>    <span class="fu"><a href="https://rdrr.io/pkg/iSEE/man/ComplexHeatmapPlot-class.html" class="external-link">ComplexHeatmapPlot</a></span><span class="op">(</span>RowSelectionSource <span class="op">=</span> <span class="st">"PathwaysTable1"</span>, </span>
<span>                       CustomRows <span class="op">=</span> <span class="cn">FALSE</span>, ColumnData <span class="op">=</span> <span class="st">"condition_name"</span>,</span>
<span>                       ClusterRows <span class="op">=</span> <span class="cn">TRUE</span>, Assay <span class="op">=</span> <span class="st">"vst"</span><span class="op">)</span>,</span>
<span>    <span class="fu"><a href="https://rdrr.io/pkg/iSEEpathways/man/FgseaEnrichmentPlot-class.html" class="external-link">FgseaEnrichmentPlot</a></span><span class="op">(</span>ResultName <span class="op">=</span> <span class="st">"IFNgTRUE.SL1344TRUE.limma.fgsea"</span>,</span>
<span>                        PathwayId <span class="op">=</span> <span class="st">"GO:0046324"</span><span class="op">)</span></span>
<span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">app</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="iseehub">iSEEhub<a class="anchor" aria-label="anchor" href="#iseehub"></a>
</h2>
<p>The <em><a href="https://bioconductor.org/packages/3.18/iSEEhub" class="external-link">iSEEhub</a></em>
package provides a custom landing page for an <em><a href="https://bioconductor.org/packages/3.18/iSEE" class="external-link">iSEE</a></em>
application interfacing with the Bioconductor <em><a href="https://bioconductor.org/packages/3.18/ExperimentHub" class="external-link">ExperimentHub</a></em>.
The landing page allows users to browse the ExperimentHub, select a data
set, download and cache it, and import it directly into an iSEE app.</p>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">ehub</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/ExperimentHub/man/ExperimentHub-class.html" class="external-link">ExperimentHub</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="va">app</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/iSEEhub/man/iSEEhub.html" class="external-link">iSEEhub</a></span><span class="op">(</span><span class="va">ehub</span><span class="op">)</span></span>
<span><span class="va">app</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="iseeindex">iSEEindex<a class="anchor" aria-label="anchor" href="#iseeindex"></a>
</h2>
<p><em><a href="https://bioconductor.org/packages/3.18/iSEEindex" class="external-link">iSEEindex</a></em>
provides an interface to any collection of data sets within a single
iSEE web-application. The main functionality of this package is to
define a custom landing page allowing app maintainers to list a custom
collection of data sets that users can selected from and directly load
objects into an iSEE web application. To see how to configure such an
app, we will create a small example:</p>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">bfc</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocFileCache/man/BiocFileCache-class.html" class="external-link">BiocFileCache</a></span><span class="op">(</span>cache <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/tempfile.html" class="external-link">tempdir</a></span><span class="op">(</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="va">dataset_fun</span> <span class="op">&lt;-</span> <span class="kw">function</span><span class="op">(</span><span class="op">)</span> <span class="op">{</span></span>
<span>    <span class="va">x</span> <span class="op">&lt;-</span> <span class="fu">yaml</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/yaml/man/read_yaml.html" class="external-link">read_yaml</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/system.file.html" class="external-link">system.file</a></span><span class="op">(</span>package <span class="op">=</span> <span class="st">"iSEEindex"</span>, <span class="st">"example.yaml"</span><span class="op">)</span><span class="op">)</span></span>
<span>    <span class="va">x</span><span class="op">$</span><span class="va">datasets</span></span>
<span><span class="op">}</span></span>
<span><span class="va">initial_fun</span> <span class="op">&lt;-</span> <span class="kw">function</span><span class="op">(</span><span class="op">)</span> <span class="op">{</span></span>
<span>    <span class="va">x</span> <span class="op">&lt;-</span> <span class="fu">yaml</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/yaml/man/read_yaml.html" class="external-link">read_yaml</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/system.file.html" class="external-link">system.file</a></span><span class="op">(</span>package <span class="op">=</span> <span class="st">"iSEEindex"</span>, <span class="st">"example.yaml"</span><span class="op">)</span><span class="op">)</span></span>
<span>    <span class="va">x</span><span class="op">$</span><span class="va">initial</span></span>
<span><span class="op">}</span></span>
<span><span class="va">app</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/iSEEindex/man/iSEEindex.html" class="external-link">iSEEindex</a></span><span class="op">(</span><span class="va">bfc</span>, <span class="va">dataset_fun</span>, <span class="va">initial_fun</span><span class="op">)</span></span>
<span><span class="va">app</span></span></code></pre></div>
<p>A more elaborate example is available at <a href="https://rehwinkellab.shinyapps.io/ifnresource/" class="external-link">https://rehwinkellab.shinyapps.io/ifnresource/</a>.
The source can be found at <a href="https://github.com/kevinrue/IFNresource" class="external-link">https://github.com/kevinrue/IFNresource</a>.</p>
</div>
<div class="section level2">
<h2 id="session-info">Session info<a class="anchor" aria-label="anchor" href="#session-info"></a>
</h2>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/sessionInfo.html" class="external-link">sessionInfo</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="co">#&gt; R version 4.3.1 (2023-06-16)</span></span>
<span><span class="co">#&gt; Platform: x86_64-pc-linux-gnu (64-bit)</span></span>
<span><span class="co">#&gt; Running under: Ubuntu 22.04.3 LTS</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Matrix products: default</span></span>
<span><span class="co">#&gt; BLAS:   /usr/lib/x86_64-linux-gnu/openblas-pthread/libblas.so.3 </span></span>
<span><span class="co">#&gt; LAPACK: /usr/lib/x86_64-linux-gnu/openblas-pthread/libopenblasp-r0.3.20.so;  LAPACK version 3.10.0</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; locale:</span></span>
<span><span class="co">#&gt;  [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              </span></span>
<span><span class="co">#&gt;  [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    </span></span>
<span><span class="co">#&gt;  [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=en_US.UTF-8   </span></span>
<span><span class="co">#&gt;  [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                 </span></span>
<span><span class="co">#&gt;  [9] LC_ADDRESS=C               LC_TELEPHONE=C            </span></span>
<span><span class="co">#&gt; [11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; time zone: Etc/UTC</span></span>
<span><span class="co">#&gt; tzcode source: system (glibc)</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; attached base packages:</span></span>
<span><span class="co">#&gt; [1] stats4    stats     graphics  grDevices utils     datasets  methods  </span></span>
<span><span class="co">#&gt; [8] base     </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; other attached packages:</span></span>
<span><span class="co">#&gt;  [1] GenomicFeatures_1.53.2      yaml_2.3.7                 </span></span>
<span><span class="co">#&gt;  [3] scater_1.29.4               ggplot2_3.4.3              </span></span>
<span><span class="co">#&gt;  [5] scuttle_1.11.2              DuoClustering2018_1.19.0   </span></span>
<span><span class="co">#&gt;  [7] GO.db_3.17.0                fgsea_1.27.1               </span></span>
<span><span class="co">#&gt;  [9] iSEEindex_0.99.13           iSEEhub_1.3.2              </span></span>
<span><span class="co">#&gt; [11] ExperimentHub_2.9.1         AnnotationHub_3.9.2        </span></span>
<span><span class="co">#&gt; [13] BiocFileCache_2.9.1         dbplyr_2.3.3               </span></span>
<span><span class="co">#&gt; [15] iSEEpathways_0.99.0         iSEEde_0.99.0              </span></span>
<span><span class="co">#&gt; [17] iSEE_2.13.5                 SingleCellExperiment_1.23.0</span></span>
<span><span class="co">#&gt; [19] stringr_1.5.0               org.Hs.eg.db_3.17.0        </span></span>
<span><span class="co">#&gt; [21] AnnotationDbi_1.63.2        tximeta_1.19.5             </span></span>
<span><span class="co">#&gt; [23] edgeR_3.43.8                limma_3.57.7               </span></span>
<span><span class="co">#&gt; [25] DESeq2_1.41.8               SummarizedExperiment_1.31.1</span></span>
<span><span class="co">#&gt; [27] Biobase_2.61.0              MatrixGenerics_1.13.1      </span></span>
<span><span class="co">#&gt; [29] matrixStats_1.0.0           GenomicRanges_1.53.1       </span></span>
<span><span class="co">#&gt; [31] GenomeInfoDb_1.37.3         IRanges_2.35.2             </span></span>
<span><span class="co">#&gt; [33] S4Vectors_0.39.1            BiocGenerics_0.47.0        </span></span>
<span><span class="co">#&gt; [35] macrophage_1.17.0           BiocStyle_2.29.1           </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; loaded via a namespace (and not attached):</span></span>
<span><span class="co">#&gt;   [1] fs_1.6.3                      ProtGenerics_1.33.1          </span></span>
<span><span class="co">#&gt;   [3] bitops_1.0-7                  httr_1.4.7                   </span></span>
<span><span class="co">#&gt;   [5] RColorBrewer_1.1-3            doParallel_1.0.17            </span></span>
<span><span class="co">#&gt;   [7] tools_4.3.1                   utf8_1.2.3                   </span></span>
<span><span class="co">#&gt;   [9] R6_2.5.1                      DT_0.29                      </span></span>
<span><span class="co">#&gt;  [11] lazyeval_0.2.2                mgcv_1.9-0                   </span></span>
<span><span class="co">#&gt;  [13] GetoptLong_1.0.5              withr_2.5.0                  </span></span>
<span><span class="co">#&gt;  [15] prettyunits_1.1.1             gridExtra_2.3                </span></span>
<span><span class="co">#&gt;  [17] cli_3.6.1                     textshaping_0.3.6            </span></span>
<span><span class="co">#&gt;  [19] shinyjs_2.1.0                 sass_0.4.7                   </span></span>
<span><span class="co">#&gt;  [21] pkgdown_2.0.7                 Rsamtools_2.17.0             </span></span>
<span><span class="co">#&gt;  [23] systemfonts_1.0.4             RSQLite_2.3.1                </span></span>
<span><span class="co">#&gt;  [25] generics_0.1.3                shape_1.4.6                  </span></span>
<span><span class="co">#&gt;  [27] BiocIO_1.11.0                 dplyr_1.1.3                  </span></span>
<span><span class="co">#&gt;  [29] Matrix_1.6-1                  ggbeeswarm_0.7.2             </span></span>
<span><span class="co">#&gt;  [31] fansi_1.0.4                   abind_1.4-5                  </span></span>
<span><span class="co">#&gt;  [33] lifecycle_1.0.3               SparseArray_1.1.12           </span></span>
<span><span class="co">#&gt;  [35] grid_4.3.1                    blob_1.2.4                   </span></span>
<span><span class="co">#&gt;  [37] promises_1.2.1                crayon_1.5.2                 </span></span>
<span><span class="co">#&gt;  [39] shinydashboard_0.7.2          miniUI_0.1.1.1               </span></span>
<span><span class="co">#&gt;  [41] lattice_0.21-8                beachmat_2.17.16             </span></span>
<span><span class="co">#&gt;  [43] cowplot_1.1.1                 KEGGREST_1.41.0              </span></span>
<span><span class="co">#&gt;  [45] pillar_1.9.0                  knitr_1.43                   </span></span>
<span><span class="co">#&gt;  [47] ComplexHeatmap_2.17.0         rjson_0.2.21                 </span></span>
<span><span class="co">#&gt;  [49] codetools_0.2-19              fastmatch_1.1-4              </span></span>
<span><span class="co">#&gt;  [51] glue_1.6.2                    data.table_1.14.8            </span></span>
<span><span class="co">#&gt;  [53] vctrs_0.6.3                   png_0.1-8                    </span></span>
<span><span class="co">#&gt;  [55] urltools_1.7.3                gtable_0.3.4                 </span></span>
<span><span class="co">#&gt;  [57] cachem_1.0.8                  xfun_0.40                    </span></span>
<span><span class="co">#&gt;  [59] S4Arrays_1.1.6                mime_0.12                    </span></span>
<span><span class="co">#&gt;  [61] iterators_1.0.14              statmod_1.5.0                </span></span>
<span><span class="co">#&gt;  [63] interactiveDisplayBase_1.39.0 ellipsis_0.3.2               </span></span>
<span><span class="co">#&gt;  [65] nlme_3.1-163                  bit64_4.0.5                  </span></span>
<span><span class="co">#&gt;  [67] progress_1.2.2                filelock_1.0.2               </span></span>
<span><span class="co">#&gt;  [69] rprojroot_2.0.3               bslib_0.5.1                  </span></span>
<span><span class="co">#&gt;  [71] irlba_2.3.5.1                 vipor_0.4.5                  </span></span>
<span><span class="co">#&gt;  [73] colorspace_2.1-0              DBI_1.1.3                    </span></span>
<span><span class="co">#&gt;  [75] tidyselect_1.2.0              bit_4.0.5                    </span></span>
<span><span class="co">#&gt;  [77] compiler_4.3.1                curl_5.0.2                   </span></span>
<span><span class="co">#&gt;  [79] BiocNeighbors_1.19.0          xml2_1.3.5                   </span></span>
<span><span class="co">#&gt;  [81] desc_1.4.2                    DelayedArray_0.27.10         </span></span>
<span><span class="co">#&gt;  [83] colourpicker_1.3.0            bookdown_0.35                </span></span>
<span><span class="co">#&gt;  [85] rtracklayer_1.61.1            triebeard_0.4.1              </span></span>
<span><span class="co">#&gt;  [87] scales_1.2.1                  rappdirs_0.3.3               </span></span>
<span><span class="co">#&gt;  [89] digest_0.6.33                 rmarkdown_2.24               </span></span>
<span><span class="co">#&gt;  [91] XVector_0.41.1                htmltools_0.5.6              </span></span>
<span><span class="co">#&gt;  [93] pkgconfig_2.0.3               sparseMatrixStats_1.13.4     </span></span>
<span><span class="co">#&gt;  [95] fastmap_1.1.1                 ensembldb_2.25.0             </span></span>
<span><span class="co">#&gt;  [97] rlang_1.1.1                   GlobalOptions_0.1.2          </span></span>
<span><span class="co">#&gt;  [99] htmlwidgets_1.6.2             ggthemes_4.2.4               </span></span>
<span><span class="co">#&gt; [101] shiny_1.7.5                   DelayedMatrixStats_1.23.4    </span></span>
<span><span class="co">#&gt; [103] jquerylib_0.1.4               jsonlite_1.8.7               </span></span>
<span><span class="co">#&gt; [105] BiocParallel_1.35.4           mclust_6.0.0                 </span></span>
<span><span class="co">#&gt; [107] BiocSingular_1.17.1           RCurl_1.98-1.12              </span></span>
<span><span class="co">#&gt; [109] magrittr_2.0.3                GenomeInfoDbData_1.2.10      </span></span>
<span><span class="co">#&gt; [111] munsell_0.5.0                 Rcpp_1.0.11                  </span></span>
<span><span class="co">#&gt; [113] viridis_0.6.4                 stringi_1.7.12               </span></span>
<span><span class="co">#&gt; [115] rintrojs_0.3.2                zlibbioc_1.47.0              </span></span>
<span><span class="co">#&gt; [117] plyr_1.8.8                    parallel_4.3.1               </span></span>
<span><span class="co">#&gt; [119] ggrepel_0.9.3                 Biostrings_2.69.2            </span></span>
<span><span class="co">#&gt; [121] splines_4.3.1                 hms_1.1.3                    </span></span>
<span><span class="co">#&gt; [123] circlize_0.4.15               locfit_1.5-9.8               </span></span>
<span><span class="co">#&gt; [125] igraph_1.5.1                  paws.common_0.5.8            </span></span>
<span><span class="co">#&gt; [127] reshape2_1.4.4                biomaRt_2.57.1               </span></span>
<span><span class="co">#&gt; [129] ScaledMatrix_1.9.1            BiocVersion_3.18.0           </span></span>
<span><span class="co">#&gt; [131] XML_3.99-0.14                 evaluate_0.21                </span></span>
<span><span class="co">#&gt; [133] BiocManager_1.30.22           foreach_1.5.2                </span></span>
<span><span class="co">#&gt; [135] httpuv_1.6.11                 tidyr_1.3.0                  </span></span>
<span><span class="co">#&gt; [137] purrr_1.0.2                   clue_0.3-64                  </span></span>
<span><span class="co">#&gt; [139] paws.storage_0.3.0            rsvd_1.0.5                   </span></span>
<span><span class="co">#&gt; [141] xtable_1.8-4                  restfulr_0.0.15              </span></span>
<span><span class="co">#&gt; [143] AnnotationFilter_1.25.0       later_1.3.1                  </span></span>
<span><span class="co">#&gt; [145] viridisLite_0.4.2             ragg_1.2.5                   </span></span>
<span><span class="co">#&gt; [147] tibble_3.2.1                  beeswarm_0.4.0               </span></span>
<span><span class="co">#&gt; [149] memoise_2.0.1                 GenomicAlignments_1.37.0     </span></span>
<span><span class="co">#&gt; [151] tximport_1.29.1               cluster_2.1.4                </span></span>
<span><span class="co">#&gt; [153] shinyWidgets_0.8.0            shinyAce_0.4.2</span></span></code></pre></div>
</div>
<div class="footnotes footnotes-end-of-document">
<hr>
<ol>
<li id="fn1"><p><a href="mailto:marinif@uni-mainz.de" class="email">marinif@uni-mainz.de</a><a href="#fnref1" class="footnote-back">↩︎</a></p></li>
<li id="fn2"><p><a href="mailto:charlottesoneson@gmail.com" class="email">charlottesoneson@gmail.com</a><a href="#fnref2" class="footnote-back">↩︎</a></p></li>
<li id="fn3"><p><a href="mailto:kevinrue67@gmail.com" class="email">kevinrue67@gmail.com</a><a href="#fnref3" class="footnote-back">↩︎</a></p></li>
</ol>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p></p>
<p>Developed by Federico Marini, Charlotte Soneson, Kevin Rue-Albrecht.</p>
</div>

<div class="pkgdown">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.7.</p>
</div>

      </footer>
</div>

  


  

  </body>
</html>
